### ISSUE TITLE
interested in testing versions of differently trained resnet50 backbone models

### ISSUE THREAD

**Issue by nikparth** [himjl/hobj#1]
Hi I found your paper very interesting and wanted to try this benchmark on some of my own pytorch models (resnet50 backbones but trained with different objectives). I was looking through your codebase and was a bit confused about how to wrap a standard backbone given the model classes you have here. Is there an example running a benchmark on a standard deep net feature extactor? Any help would be greatly appreciated!

**Comment by himjl** [comment]
Hi Nikhil, 

Thank you very much for your interest in my work!

If you would like to wrap a pretrained feature extractor, a natural place to do so would be by implementing a new RepresentationalModel <https://github.com/himjl/hobj/blob/master/hobj/learning_models/linear_learners/representational_models/representational_model.py> class -- e.g. using the PrecachedRepresentationalModel. This simply requires supplying a url_to_features dictionary to the constructor. 

If you'd like to use a linear learning rule as discussed in the paper to learn "on top of" this representational model, you may then construct a LinearLearner <https://github.com/himjl/hobj/blob/master/hobj/learning_models/linear_learners/linear_learner.py> instance,  passing in that RepresentationalModel in as one of the required arguments. 

For the update_rule, you can use one of the 7 learning rules that are already implemented here <https://github.com/himjl/hobj/blob/master/hobj/learning_models/linear_learners/update_rules/update_rules.py>, or create a new learning rule which is a subclass of UpdateRule. 

Once you have that new LinearLearner instance created, you can score it on the two learning benchmarks in the paper by following the example here <https://github.com/himjl/hobj/blob/master/examples/score_example_model.py>.


Happy to hop on a Zoom call if you have any remaining questions. 

Sincerely,
Michael

**Comment by nikparth** [comment]
Ok great thanks! I'll try to do this and let you know if I have any issues.

Best,
Nikhil

**Comment by nikparth** [comment]
Hi Michael,
So I have this running but I'm getting some pretty poor results with ResNet-50 (average pool layer) supervised pre-trained (MSEn for the high-var benchmark of around 0.09, spearman consistency 0.32 and overall late accuracy of ~ 0.6). Is this accurate? there aren't any specific numbers for models in the paper so if you either had the numbers to make those plots or could tell me if these numbers for the resnet50 are accurate that would be great. This is with a linear learner with update rule (square) and I swept the learning rate alpha and didn't see much difference. 

Thanks! 
Nikhil

**Comment by nikparth** [comment]
Also do you have the numbers to create the main plots? I'd love to figure out how to plot my custom model performance against the models you tested without having to re-run all of them. 

Thanks!
